cosmo/monitors/nuv_dark_monitor.py

```diff
@@ -0,0 +1,230 @@
+            subdf = pd.DataFrame({
+                "EXPSTART": row["EXPSTART"], "TIME": [row["TIME"]],
+                "XCORR": [row["XCORR"]], "YCORR": [row["YCORR"]],
+                "TIME_3": [row["TIME_3"]]
+                })
+
+            # this is temporary until i understand james' suggestion to use
+            # df.apply and lambda functions
+            xcorr = subdf["XCORR"][0]
+            ycorr = subdf["YCORR"][0]
+            filtered_xcorr = xcorr[
+                np.where((xcorr > xlim[0]) & (xcorr < xlim[1]))]
+            filtered_ycorr = ycorr[
+                np.where((ycorr > ylim[0]) & (ycorr < ylim[1]))]
+            subdf["XCORR"] = [filtered_xcorr]
+            subdf["YCORR"] = [filtered_ycorr]
+
+            dark_rate_array, dec_year_array = self.calculate_dark_rate(subdf,
+                                                                       xlim,
+                                                                       ylim)
+            dark_rate_column.append(dark_rate_array)
+            dec_year_column.append(dec_year_array)
+
+        data["DARK_RATE"] = dark_rate_column
+        data["DECIMAL_YEAR"] = dec_year_column
+
+        # when the monitor method of the monitor is called, it will
+        # initialize the self.data attribute
+        # with this method and then can be used by the other methods
+        return data
+
+    def calculate_dark_rate(self, dataframe, xlim, ylim):
+        # calculate dark rate for one exposure, with a dataframe with TIME,
+        # XCORR, and YCORR values
+
+        # need to set this somewhere
+        timestep = 25
+        time_bins = dataframe["TIME_3"][0][::timestep]
+
+        counts = np.histogram(dataframe["TIME_3"][0], bins=time_bins)[0]
+        npix = float((xlim[1] - xlim[0]) * (ylim[1] - ylim[0]))
+        dark_rate_array = counts / npix / timestep
+        # save the whole histogram in time bins, and then plot each of them
+
+        # make a decimal year array corresponding to the time bins of the
+        # dark rates
+        # do this with the expstart (mjd) and time array from the timeline
+        # extension
+        # taking the expstart, binning the time array by the timestep,
+        # removing the last element in the array (bin by front edge),
+        # and then multiplying by the conversion factor
+        # this is done by the absolute_time helper function
+        mjd_array = absolute_time(expstart=dataframe['EXPSTART'][0],
+                                  time=dataframe['TIME_3'][0][::timestep][:-1])
+        dec_year_array = mjd_array.decimalyear
+
+        return dark_rate_array, dec_year_array
+
+    def track(self):
+        # track something. perhaps current dark rate?
+        if self.data is None:
+            self.data = self.get_data()
+
+        plotdf = pd.DataFrame({
+            "DECIMAL_YEAR": self.data["DECIMAL_YEAR"],
+            "DARK_RATE": self.data["DARK_RATE"]
+            })
```

Comment:
```suggestion
        plotdf = self.data[['DECIMAL_YEAR', 'DARK_RATE']]
```
No need to create a new DataFrame here, just slice. Or just use `self.data` directly ?